addon/model-data.js

```diff
@@ -17,10 +18,38 @@ function commitDataAndNotify(modelData, updates) {
 class M3SchemaInterface {
   constructor(modelData) {
     this.modelData = modelData;
+    this._keyBeingResolved = null;
+    this._refKeyUnknownKeysMap = {};
+  }
+
+  _beginUnknownKeyResolution(key) {
+    assert('_keyBeingResolved should be null.', this._keyBeingResolved === null);
+    this._keyBeingResolved = key;
+  }
+
+  _endUnknownKeyResolution(key) {
+    assert('_keyBeingResolved should be set.', key && this._keyBeingResolved === key);
```

Comment:
same comment here as prior assert

if there's a user-error that can wind up here let's give a more user-specific message.  If we can't think of a user-error that could get them here then leaving an internal assertion message is 👍